src/pages/core-concepts/astro-components.md

```diff
@@ -3,151 +3,206 @@ layout: ~/layouts/Main.astro
 
-```jsx
+#### Dynamic Values
+
+```astro
 ---
-let name = 'world';
+const name = "Your name here";
 ---
+<div>
+  <h1>Hello {name}!</h1>
+</div>
+```
+#### Dynamic Attributes
 
-<main>
-    <h1>Hello {name}!</h1>
-</main>
+```astro
+---
+const name = "Your name here";
+---
+<div>
+  <div data-name={name}>Attribute expressions supported</div>
+  <div data-hint={`Use JS template strings to mix ${"variables"}.`}>So good!</div>
+</div>
 ```
 
-`.astro` components can also accept props when they are rendered. Public props are exposed on the `Astro.props` global.
+#### Dynamic HTML
 
-```jsx
+```astro
 ---
-const { greeting = 'Hello', name } = Astro.props;
+const items = ["Dog", "Cat", "Platipus"];
 ---
+<ul>
+  {items.map((item) => (
+    <li>{item}</li>
+  ))}
+</ul>
+```
+
+
+### Component Props
+
+An Astro component can define and accept props. Props are available on the `Astro.props` global in your frontmatter script.
 
-<main>
-    <h1>{greeting} {name}!</h1>
-</main>
+```astro
+---
+// Example: <SomeComponent greeting="(Optional) Hello" name="Required Name" />
+const { greeting = 'Hello', name } = Astro.props;
+---
+<div>
+    <h1>{greeting}, {name}!</h1>
+</div>
 ```
 
-To define the props which your component accepts, you may export a TypeScript interface or type named `Props`.
+You can define your props with TypeScript by exporting a `Props` type interface. Astro will automatically pick up any exported `Props` interface and give type warnings/errors for your project.
 
-```tsx
+```astro
 ---
+// Example: <SomeComponent />  (WARNING: "name" prop is required)
 export interface Props {
   name: string;
   greeting?: string;
 }
-
 const { greeting = 'Hello', name } = Astro.props;
 ---
-
-<main>
-    <h1>{greeting} {name}!</h1>
-</main>
+<div>
+    <h1>{greeting}, {name}!</h1>
+</div>
 ```
 
-### Fragments
+### Fragments & Multiple Elements
 
-At the top-level of an `.astro` file, you may render any number of elements.
+At the top-level of an `.astro` file, you are safe to render as many top-level elements as you'd like. This is possible because HTML makes it easy. 
 
 ```html
-<!-- Look, no Fragment! -->
+<!-- An Astro component can render multiple HTML elements: -->
 <div id="a" />
 <div id="b" />
 <div id="c" />
 ```
 
-Inside of an expression, you must wrap multiple elements in a Fragment. Fragments must open with `<>` and close with `</>`.
+Inside of a JSX expression, you must wrap multiple elements inside of a Fragment. Fragments let you group a list of children without adding extra nodes to the DOM. This is due to a limitation of JavaScript: You can never `return` more than one thing from an expression or function. 
 
-```jsx
-<div>
-  {[0, 1, 2].map((id) => (
+A Fragment must open with `<>` and close with `</>`. Don't worry if you forget this, Astro's compiler will warn you that you need to add one.
```

Comment:
I'm not sure if we _can_ warn if you need to add a slot. Regardless, we should open an issue to track this.

Reply:
https://github.com/snowpackjs/astro/issues/600